flink-table-store-core/src/main/java/org/apache/flink/table/store/file/casting/CastedRowData.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.store.file.casting;
+
+import org.apache.flink.table.data.ArrayData;
+import org.apache.flink.table.data.DecimalData;
+import org.apache.flink.table.data.MapData;
+import org.apache.flink.table.data.RawValueData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.StringData;
+import org.apache.flink.table.data.TimestampData;
+import org.apache.flink.types.RowKind;
+
+/**
+ * An implementation of {@link RowData} which provides a casted view of the underlying {@link
+ * RowData}.
+ *
+ * <p>It reads data from underlying {@link RowData} according to source logical type and casts it
+ * with specific {@link CastExecutor}.
+ *
+ * <p>Note: This class supports only top-level castings, not nested castings.
+ */
+public class CastedRowData implements RowData {
+
+    private final FieldGetterCastExecutor[] castMapping;
```

Comment:
We need `FieldGetter` to get object value from `InternalRow` in `CastedRow` for `CastExecutor`, otherwise, we need to call getByte/getInt/getLong/getDouble and .etc method to get the underlying data value in `InternalRow` with given pos